### ISSUE TITLE
Error when generating in RuinedFooocus

### ISSUE THREAD

**Issue by Dr-KerbalMD** [LykosAI/StabilityMatrix#331]
Windows 11, RTX 3090, Stability Matrix 2.7.1.

```
Exception in thread Thread-2 (worker):
Traceback (most recent call last):
  File "threading.py", line 1016, in _bootstrap_inner
  File "threading.py", line 953, in run
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\modules\async_worker.py", line 311, in worker
    handler(task)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\modules\async_worker.py", line 56, in handler
    process(gen_data)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\modules\async_worker.py", line 225, in process
    imgs = pipeline.process(
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\modules\sdxl_pipeline.py", line 519, in process
    samples = sampler.sample(noise, positive_copy, negative_copy, **kwargs)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\samplers.py", line 708, in sample
    return sample(self.model, noise, positive, negative, cfg, self.device, sampler, sigmas, self.model_options, latent_image=latent_image, denoise_mask=denoise_mask, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\samplers.py", line 614, in sample
    samples = sampler.sample(model_wrap, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\samplers.py", line 553, in sample
    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\k_diffusion\sampling.py", line 701, in sample_dpmpp_2m_sde_gpu
    return sample_dpmpp_2m_sde(model, x, sigmas, extra_args=extra_args, callback=callback, disable=disable, eta=eta, s_noise=s_noise, noise_sampler=noise_sampler, solver_type=solver_type)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\k_diffusion\sampling.py", line 615, in sample_dpmpp_2m_sde
    callback({'x': x, 'i': i, 'sigma': sigmas[i], 'sigma_hat': sigmas[i], 'denoised': denoised})
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\repositories\ComfyUI-from-StabilityAI-Official\comfy\samplers.py", line 548, in <lambda>
    k_callback = lambda x: callback(x["i"], x["denoised"], x["x"], total_steps)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\modules\sdxl_pipeline.py", line 468, in callback_function
    callback(step, x0, x, total_steps, y)
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\modules\async_worker.py", line 194, in callback
    shared.state["preview_grid"].save(
  File "F:\StabilityMatrix-win-x64\Data\Packages\RuinedFooocus\venv\lib\site-packages\PIL\Image.py", line 2317, in save
    fp = builtins.open(filename, "w+b")
FileNotFoundError: [Errno 2] No such file or directory: 'F:\\StabilityMatrix-win-x64\\Data\\Packages\\RuinedFooocus\\modules\\..\\outputs\\preview.jpg'
```

Similar but slightly different error being reported in the RuinedFooocus repo itself:

https://github.com/runew0lf/RuinedFooocus/issues/92

**Comment by drdancm** [comment]
Issue with Ruined Fooocus. Here is the link to what we reported on that site:

https://github.com/runew0lf/RuinedFooocus/issues/92

**Comment by ionite34** [comment]
Seems to be a missing "outputs" folder in the package root, if you make that folder manually it should work. Not sure if they'll fix upstream but we'll add a folder create during install in next update.

**Comment by mohnjiles** [comment]
Should be fixed as of v2.7.2. Feel free to reopen if the issue isn't resolved. Thanks!
